# 1. Import libraries

In [36]:
import pandas as pd
from pathlib import Path
from collections import Counter

# 2. EDA

In [13]:
# Check out what is in train.csv
train_df = pd.read_csv("../data/global-wheat-detection/train.csv")
print(f"train.csv shape {train_df.shape}")
train_df.head()

train.csv shape (147793, 5)


image_id  width  height                         bbox   source
0  b6ab77fd7   1024    1024   [834.0, 222.0, 56.0, 36.0]  usask_1
1  b6ab77fd7   1024    1024  [226.0, 548.0, 130.0, 58.0]  usask_1
2  b6ab77fd7   1024    1024  [377.0, 504.0, 74.0, 160.0]  usask_1
3  b6ab77fd7   1024    1024  [834.0, 95.0, 109.0, 107.0]  usask_1
4  b6ab77fd7   1024    1024  [26.0, 144.0, 124.0, 117.0]  usask_1

In [12]:
# Look at unique values in each column
for col in train_df.columns.to_list():
    display(train_df[col].value_counts())

image_id
35b935b6c    116
f1794c924    108
f79336a8e    104
134d4a01c    101
4cf91995d     97
            ... 
86296fc32      1
ce4124be0      1
4e6c05213      1
a5cb30c38      1
41457a646      1
Name: count, Length: 3373, dtype: int64

width
1024    147793
Name: count, dtype: int64

height
1024    147793
Name: count, dtype: int64

bbox
[972, 0, 52, 84]              2
[0.0, 957.0, 79.0, 66.0]      2
[372.0, 856.0, 66.0, 64.0]    2
[302.0, 815.0, 64.0, 75.0]    2
[0.0, 385.0, 56.0, 46.0]      2
                             ..
[0, 750, 39, 78]              1
[121, 931, 119, 77]           1
[279, 965, 91, 59]            1
[588, 953, 81, 71]            1
[875.0, 740.0, 94.0, 61.0]    1
Name: count, Length: 147761, dtype: int64

source
ethz_1       51489
arvalis_1    45716
rres_1       20236
arvalis_3    16665
usask_1       5807
arvalis_2     4179
inrae_1       3701
Name: count, dtype: int64

```Observations```

1. 3373 images, all 1024 x 1024 px
2. Probably can disregard source
3. I will need to draw the bounding boxes into each image id

In [39]:
# Check number of images in train and test folder
train_dir = "../data/global-wheat-detection/train"
test_dir = "../data/global-wheat-detection/test"

train_img = [image for image in Path(train_dir).iterdir()]
train_img_ext_count = Counter([image.suffix for image in train_img])
print(f"Images in train folder: {len(train_img)}")
print(f"Image suffixes: {train_img_ext_count}")

print()

test_img = [image for image in Path(test_dir).iterdir()]
test_img_ext_count = Counter([image.suffix for image in test_img])
print(f"Images in test folder: {len(test_img)}")
print(f"Image suffixes: {test_img_ext_count}")

Images in train folder: 3422
Image suffixes: Counter({'.jpg': 3422})

Images in test folder: 10
Image suffixes: Counter({'.jpg': 10})


```Observations```

1. There are 3422 images in train folder, but only 3733 unique IDs in train.csv.
    - Explore further to see which unique IDs do not have images in train folder
2. All .jpg extension in train folder. Great

In [ ]:
# Investigate missing IDs in train.csv
